packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -642,56 +730,187 @@ abstract class TableBase extends Resource implements ITable {
+   * the tables in the account and region the table resides in.
    */
   public metricUserErrors(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
-    return this.metric('UserErrors', { statistic: 'sum', ...props });
+
+    if (props?.dimensions) {
+      throw new Error('"dimensions" is not supported for the "UserErrors" metric');
+    }
+
+    // overriding 'dimensions' here because this metric is an account metric.
+    // see 'UserErrors' in https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/metrics-dimensions.html
+    return this.metric('UserErrors', { statistic: 'sum', ...props, dimensions: {} });
   }
 
   /**
    * Metric for the conditional check failed requests this table
-   *
-   * @default sum over a minute
    */
   public metricConditionalCheckFailedRequests(props?: cloudwatch.MetricOptions): cloudwatch.Metric {
     return this.metric('ConditionalCheckFailedRequests', { statistic: 'sum', ...props });
   }
 
   /**
-   * Metric for the successful request latency this table
+   * Metric for the successful request latency for specific operations of this table
+   */
+  public metricSuccessfulRequestLatencyForOperations(props: OperationalSuccessfulRequestLatencyMetricOptions): cloudwatch.IMetric {
+
+    if (props?.dimensions?.Operation) {
+      // for simplicity, lets not support two different properties representing the same thing.
+      // note that `operations` property is required for this method.
+      throw new Error("The Operation dimension is not supported. Use the 'operations' property.");
+    }
+
+    if (props.operations.length > 5 || props.operations.length === 0) {
+      // you must pass at least 1 operation since this is an operation dimensionality metric.
+      // you cannot pass more than 5 because this would make this metric unusable for alarms, who have a 10 individual metric limit for math expressions.
+      // https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarms-on-metric-math-expressions
+      throw new Error("Number of operations for the 'metricSuccessfulRequestLatency' metric should be between 1 and 5");
+    }
+
+    // create mappers to control the metric names in the metric map since we
+    // need to need them for correctly calculating the numerator.
+    const valueMetricNameMapper = (op: Operation) => `${op}Value`.toLowerCase();
+    const countMetricNameMapper = (op: Operation) => `${op}Count`.toLowerCase();
+
+    const values = this.createMetricsForOperations('SuccessfulRequestLatency', props.operations, props, valueMetricNameMapper);
+
+    // we override the statistic here because we always need the count for a weighted average.
+    // the statistic passed by the user is considered in the value metric.
+    const counts = this.createMetricsForOperations('SuccessfulRequestLatency', props.operations, { ...props, statistic: 'n' }, countMetricNameMapper);
```

Comment:
This is a little inefficient, iterating twice over the same array. But makes the code much more readable in my opinion, and i'm perfectly happy to sacrifice one for the other.

Reply:
I'm not convinced that `N * (do 2 operations)` is slower than `2 * (do N operations)`

Reply:
negligible is an understatement